File: disc_golf_app/lib/screens/knowledge_base/category_screen.dart
import 'package:flutter/material.dart';
import 'package:provider/provider.dart';
import '../../models/knowledge_base.dart';
import '../../services/knowledge_base_service.dart';
import 'article_detail_screen.dart';

class CategoryScreen extends StatelessWidget {
  final KBCategory category;

  const CategoryScreen({Key? key, required this.category}) : super(key: key);

  @override
  Widget build(BuildContext context) {
    final service = Provider.of<KnowledgeBaseService>(context);
    final tips = service.getArticles(category: category.id, type: 'tip');
    final faqs = service.getArticles(category: category.id, type: 'faq');
    final color = Color(category.colorValue);

    return Scaffold(
      appBar: AppBar(
        title: Text(category.name),
      ),
      body: SingleChildScrollView(
        padding: const EdgeInsets.all(16),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text(
              category.description,
              style: TextStyle(color: Colors.grey.shade400, fontSize: 14),
            ),
            if (tips.isNotEmpty) ...[
              const SizedBox(height: 20),
              Text(
                'Tips',
                style: Theme.of(context).textTheme.titleMedium?.copyWith(
                      fontWeight: FontWeight.bold,
                    ),
              ),
              const SizedBox(height: 8),
              ...tips.map((tip) => _buildTipCard(context, tip, service, color)),
            ],
            if (faqs.isNotEmpty) ...[
              const SizedBox(height: 24),
              Text(
                'Frequently Asked Questions',
                style: Theme.of(context).textTheme.titleMedium?.copyWith(
                      fontWeight: FontWeight.bold,
                    ),
              ),
              const SizedBox(height: 8),
              ...faqs.map((faq) => _buildFaqTile(context, faq, service, color)),
            ],
          ],
        ),
      ),
    );
  }

  Widget _buildTipCard(
    BuildContext context,
    KBArticle tip,
    KnowledgeBaseService service,
    Color color,
  ) {
    final source =
        tip.sourceIds.isNotEmpty ? service.getStudy(tip.sourceIds.first) : null;

    return Card(
      margin: const EdgeInsets.only(bottom: 8),
      color: color.withAlpha(25),
      shape: RoundedRectangleBorder(
        borderRadius: BorderRadius.circular(8),
        side: BorderSide(color: color.withAlpha(50)),
      ),
      child: Padding(
        padding: const EdgeInsets.all(14),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Row(
              children: [
                Icon(Icons.lightbulb, color: Colors.amber.shade300, size: 18),
                const SizedBox(width: 8),
                Expanded(
                  child: Text(
                    tip.question,
                    style: TextStyle(
                      fontWeight: FontWeight.bold,
                      color: color,
                      fontSize: 13,
                    ),
                  ),
                ),
              ],
            ),
            const SizedBox(height: 8),
            Text(
              tip.answer,
              style: const TextStyle(fontSize: 13, height: 1.4),
            ),
            if (source != null) ...[
              const SizedBox(height: 6),
              Text(
                '— ${source.citation}',
                style: TextStyle(
                  fontSize: 11,
                  color: Colors.grey.shade500,
                  fontStyle: FontStyle.italic,
                ),
              ),
            ],
          ],
        ),
      ),
    );
  }

  Widget _buildFaqTile(
    BuildContext context,
    KBArticle faq,
    KnowledgeBaseService service,
    Color color,
  ) {
    return Card(
      margin: const EdgeInsets.only(bottom: 4),
      child: ListTile(
        title: Text(
          faq.question,
          style: const TextStyle(fontSize: 14, fontWeight: FontWeight.w500),
        ),
        trailing: Icon(Icons.chevron_right, color: color),
        onTap: () => Navigator.push(
          context,
          MaterialPageRoute(
            builder: (_) => ArticleDetailScreen(article: faq),
          ),
        ),
      ),
    );
  }
}

File: disc_golf_android/app/src/main/kotlin/com/discflightschool/app/ui/screens/knowledge/CategoryScreen.kt
package com.discflightschool.app.ui.screens.knowledge

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.Spacer
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.layout.width
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.automirrored.filled.MenuBook
import androidx.compose.material.icons.filled.ChevronRight
import androidx.compose.material3.Icon
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Scaffold
import androidx.compose.material3.Surface
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import com.discflightschool.app.LocalAppContainer
import com.discflightschool.app.ui.components.AppTopBar
import com.discflightschool.app.ui.components.LoadingState
import com.discflightschool.app.ui.theme.AppColors
import com.discflightschool.core.model.KBArticle
import com.discflightschool.core.model.KBStudy

/** Everything the library has on one topic: tips first, then the FAQs. */
@Composable
fun CategoryScreen(
    categoryId: String,
    onBack: () -> Unit,
    onOpenArticle: (String) -> Unit,
) {
    val container = LocalAppContainer.current
    val repository = container.knowledgeBaseRepository

    LaunchedEffect(Unit) { repository.load() }

    val category = repository.categories.firstOrNull { it.id == categoryId }
    if (category == null) {
        Scaffold(topBar = { AppTopBar(title = "Category", onBack = onBack) }) { padding ->
            LoadingState("Loading...", Modifier.padding(padding))
        }
        return
    }

    val color = Color(category.colorValue)
    val tips = repository.articles(category = categoryId, type = "tip")
    val faqs = repository.articles(category = categoryId, type = "faq")

    Scaffold(topBar = { AppTopBar(title = category.name, onBack = onBack) }) { padding ->
        Column(
            modifier = Modifier
                .padding(padding)
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
        ) {
            Text(
                category.description,
                style = MaterialTheme.typography.bodyMedium,
                color = AppColors.Muted,
            )

            if (tips.isNotEmpty()) {
                Spacer(Modifier.height(20.dp))
                Text(
                    "Tips",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                )
                Spacer(Modifier.height(8.dp))
                tips.forEach { tip ->
                    TipCard(
                        tip = tip,
                        citation = tip.sourceIds.firstOrNull()
                            ?.let { repository.study(it)?.citation },
                        accent = color,
                    )
                    Spacer(Modifier.height(8.dp))
                }
            }

            if (faqs.isNotEmpty()) {
                Spacer(Modifier.height(16.dp))
                Text(
                    "Frequently asked questions",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                )
                Spacer(Modifier.height(8.dp))
                faqs.forEach { faq ->
                    FaqRow(faq, color) { onOpenArticle(faq.id) }
                    Spacer(Modifier.height(4.dp))
                }
            }

            Spacer(Modifier.height(24.dp))
        }
    }
}

@Composable
private fun FaqRow(article: KBArticle, accent: Color, onClick: () -> Unit) {
    Surface(
        color = AppColors.Surface,
        shape = RoundedCornerShape(8.dp),
        modifier = Modifier.fillMaxWidth(),
    ) {
        Row(
            modifier = Modifier
                .clickable(onClick = onClick)
                .padding(horizontal = 16.dp, vertical = 14.dp),
            verticalAlignment = Alignment.CenterVertically,
        ) {
            Text(
                article.question,
                modifier = Modifier.weight(1f),
                style = MaterialTheme.typography.bodyMedium,
                fontWeight = FontWeight.Medium,
            )
            Icon(Icons.Default.ChevronRight, contentDescription = null, tint = accent)
        }
    }
}

/** One FAQ in full, with its key findings and the studies behind it. */
@Composable
fun ArticleDetailScreen(articleId: String, onBack: () -> Unit) {
    val container = LocalAppContainer.current
    val repository = container.knowledgeBaseRepository

    LaunchedEffect(Unit) { repository.load() }

    val article = repository.articles.firstOrNull { it.id == articleId }
    if (article == null) {
        Scaffold(topBar = { AppTopBar(title = "FAQ", onBack = onBack) }) { padding ->
            LoadingState("Loading...", Modifier.padding(padding))
        }
        return
    }

    Scaffold(topBar = { AppTopBar(title = "FAQ detail", onBack = onBack) }) { padding ->
        Column(
            modifier = Modifier
                .padding(padding)
                .fillMaxSize()
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
        ) {
            Text(
                article.question,
                style = MaterialTheme.typography.titleLarge,
                fontWeight = FontWeight.Bold,
            )
            Spacer(Modifier.height(16.dp))
            Text(article.answer, style = MaterialTheme.typography.bodyLarge)

            if (article.keyFindings.isNotEmpty()) {
                Spacer(Modifier.height(20.dp))
                Text(
                    "Key findings",
                    style = MaterialTheme.typography.titleMedium,
                    fontWeight = FontWeight.Bold,
                )
                Spacer(Modifier.height(8.dp))
                article.keyFindings.forEach { finding ->
                    Row(Modifier.padding(bottom = 8.dp)) {
                        Text("  •  ", fontWeight = FontWeight.Bold)
                        Text(finding, style = MaterialTheme.typography.bodyMedium)
                    }
                }
            }

            val sources = article.sourceIds.mapNotNull { repository.study(it) }
            if (sources.isNotEmpty()) {
                Spacer(Modifier.height(24.dp))
                Row(verticalAlignment = Alignment.CenterVertically) {
                    Icon(
                        Icons.AutoMirrored.Filled.MenuBook,
                        contentDescription = null,
                        tint = AppColors.KnowledgeBase,
                        modifier = Modifier.size(20.dp),
                    )
                    Spacer(Modifier.width(8.dp))
                    Text(
                        "Sources",
                        style = MaterialTheme.typography.titleMedium,
                        fontWeight = FontWeight.Bold,
                    )
                }
                Spacer(Modifier.height(8.dp))
                sources.forEach { study ->
                    SourceCard(study)
                    Spacer(Modifier.height(8.dp))
                }
            }

            Spacer(Modifier.height(24.dp))
        }
    }
}

@Composable
private fun SourceCard(study: KBStudy) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(
                AppColors.KnowledgeBase.copy(alpha = 0.12f),
                RoundedCornerShape(8.dp),
            )
            .border(
                1.dp,
                AppColors.KnowledgeBase.copy(alpha = 0.25f),
                RoundedCornerShape(8.dp),
            )
            .padding(12.dp),
    ) {
        Text(
            study.citation,
            fontWeight = FontWeight.Bold,
            color = AppColors.KnowledgeBase,
            style = MaterialTheme.typography.bodyMedium,
        )
        Spacer(Modifier.height(4.dp))
        Text(study.title, style = MaterialTheme.typography.bodyMedium)
        Spacer(Modifier.height(6.dp))
        Text(
            study.summary,
            style = MaterialTheme.typography.bodySmall,
            color = AppColors.Muted,
        )
        Spacer(Modifier.height(4.dp))
        Text(
            study.filename,
            style = MaterialTheme.typography.labelSmall,
            color = AppColors.Muted,
            fontStyle = FontStyle.Italic,
        )
    }
}
